flytekit/core/task.py

```diff
@@ -343,10 +344,16 @@ def launch_dynamically():
     """
 
     def wrapper(fn: Callable[P, Any]) -> PythonFunctionTask[T]:
+        if isinstance(cache, list) and all(isinstance(item, AutoCache) for item in cache):
+            cache_versions = [item.get_version() for item in cache]
+            task_hash = "".join(cache_versions)
```

Comment:
@eapolinario does the cache version string have a character limit? If so we may need to re-hash the concatenated hashes.

Reply:
Ah I had a feeling this would be an issue. I just left the long join for simplicity and to not introduce a hashing function in `flytekit/core/task.py` , but happy to hash that.